core/java/com/google/instrumentation/common/MatchFunction.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.common;
+
+/**
+ * Encodes functions {@code func: A -> B} for use in encoding disjoint-unions.
```

Comment:
Another option is to use visitor pattern terminology and have an `AggregationDescriptorVisitor` interface with two methods inside `AggregationDescriptor`.  The downside is that users wouldn't be able to take advantage of lambdas with Java 8.